File: Nx3CS/bin/test/src/nx3/PHeadsRectsCalculator.cs

#pragma warning disable 109, 114, 219, 429, 168, 162
namespace nx3{
	public  class PHeadsRectsCalculator : global::haxe.lang.HxObject {
		public    PHeadsRectsCalculator(global::haxe.lang.EmptyObject empty){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				{
				}
				
			}
			#line default
		}
		
		
		public    PHeadsRectsCalculator(global::nx3.PNote note, global::nx3.EDirectionUD direction){
			unchecked {
				#line 22 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				global::nx3.PHeadsRectsCalculator.__hx_ctor_nx3_PHeadsRectsCalculator(this, note, direction);
			}
			#line default
		}
		
		
		public static   void __hx_ctor_nx3_PHeadsRectsCalculator(global::nx3.PHeadsRectsCalculator __temp_me104, global::nx3.PNote note, global::nx3.EDirectionUD direction){
			unchecked {
				#line 24 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				if (( direction != default(global::nx3.EDirectionUD) )) {
					#line 24 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
					__temp_me104.direction = direction;
				}
				 else {
					#line 24 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
					__temp_me104.direction = note.getDirection();
				}
				
				#line 25 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				__temp_me104.vheads = note.get_heads();
				__temp_me104.placements = new global::nx3.PHeadPlacementsCalculator(((global::Array<object>) (__temp_me104.vheads) ), ((global::nx3.EDirectionUD) (__temp_me104.direction) )).getHeadsPlacements();
				__temp_me104.notevalue = note.nnote.@value;
			}
			#line default
		}
		
		
		public static  new object __hx_createEmpty(){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				return new global::nx3.PHeadsRectsCalculator(((global::haxe.lang.EmptyObject) (global::haxe.lang.EmptyObject.EMPTY) ));
			}
			#line default
		}
		
		
		public static  new object __hx_create(global::Array arr){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				return new global::nx3.PHeadsRectsCalculator(((global::nx3.PNote) (arr[0]) ), ((global::nx3.EDirectionUD) (arr[1]) ));
			}
			#line default
		}
		
		
		public  global::Array<object> vheads;
		
		public  global::Array<object> placements;
		
		public  global::nx3.ENoteVal notevalue;
		
		public  global::nx3.ENoteType notetype;
		
		public  global::nx3.EDirectionUD direction;
		
		public virtual   global::Array<object> getHeadsRects(){
			unchecked {
				#line 32 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				global::Array<object> rects = new global::Array<object>();
				#line 36 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				int i = 0;
				{
					#line 37 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
					int _g = 0;
					#line 37 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
					global::Array<object> _g1 = this.placements;
					#line 37 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
					while (( _g < _g1.length )){
						#line 37 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						object placement = _g1[_g];
						#line 37 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						 ++ _g;
						#line 39 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						global::nx3.EHeadType headtype = ((global::nx3.PHead) (this.vheads[i]) ).nhead.type;
						#line 41 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						global::nx3.geom.Rectangle rect = default(global::nx3.geom.Rectangle);
						double headw = ((double) (0) );
						#line 44 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						this.headRect(headtype, this.notevalue);
						#line 47 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						{
							#line 47 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
							global::nx3.EHeadValueType _g2 = global::nx3.ENoteValTools.head(this.notevalue);
							#line 47 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
							switch (global::Type.enumIndex(_g2)){
								case 2:
								{
									#line 50 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									headw = 2.2;
									#line 50 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									break;
								}
								
								
								default:
								{
									#line 52 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									headw = 1.6;
									#line 52 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									break;
								}
								
							}
							
						}
						
						#line 55 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						global::nx3.geom.Rectangle rect1 = new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>( - (headw) , true), new global::haxe.lang.Null<double>(((double) (-1) ), true), new global::haxe.lang.Null<double>(( 2 * headw ), true), new global::haxe.lang.Null<double>(((double) (2) ), true));
						#line 57 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						double pos = 0.0;
						{
							#line 58 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
							global::nx3.EHeadPosition _g21 = ((global::nx3.EHeadPosition) (global::haxe.lang.Runtime.getField(placement, "pos", 5594516, true)) );
							#line 58 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
							switch (global::Type.enumIndex(_g21)){
								case 0:
								{
									#line 60 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									pos = ( -2 * headw );
									#line 60 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									break;
								}
								
								
								case 2:
								{
									#line 61 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									pos = ( 2 * headw );
									#line 61 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									break;
								}
								
								
								default:
								{
									#line 62 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									pos = ((double) (0) );
									#line 62 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
									break;
								}
								
							}
							
						}
						
						#line 64 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						rect1.offset(pos, ((double) (((int) (global::haxe.lang.Runtime.getField_f(placement, "level", 1919096196, true)) )) ));
						rects.push(rect1);
						i++;
					}
					
				}
				
				#line 69 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				return rects;
			}
			#line default
		}
		
		
		public virtual   global::nx3.geom.Rectangle headRect(global::nx3.EHeadType type, global::nx3.ENoteVal notevalue){
			unchecked {
				#line 74 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				int headw = 2;
				int headh = 2;
				switch (global::Type.enumIndex(type)){
					case 0:
					{
						#line 79 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						global::nx3.EHeadValueType _g = global::nx3.ENoteValTools.head(this.notevalue);
						#line 79 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						switch (global::Type.enumIndex(_g)){
							case 2:
							{
								#line 82 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
								return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(-2.2, true), new global::haxe.lang.Null<double>(((double) (-1) ), true), new global::haxe.lang.Null<double>(4.4, true), new global::haxe.lang.Null<double>(((double) (2) ), true));
							}
							
							
							default:
							{
								#line 84 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
								return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(-1.6, true), new global::haxe.lang.Null<double>(((double) (-1) ), true), new global::haxe.lang.Null<double>(3.2, true), new global::haxe.lang.Null<double>(((double) (2) ), true));
							}
							
						}
						
					}
					
					
					case 3:
					{
						#line 87 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						int _g1 = global::nx3.ENoteValTools.beaminglevel(this.notevalue);
						#line 87 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						switch (_g1){
							case 1:
							{
								#line 89 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
								return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(-1.8, true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(3.6, true), new global::haxe.lang.Null<double>(((double) (6) ), true));
							}
							
							
							case 2:
							{
								#line 90 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
								return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (-2) ), true), new global::haxe.lang.Null<double>(((double) (-3) ), true), new global::haxe.lang.Null<double>(((double) (4) ), true), new global::haxe.lang.Null<double>(((double) (6) ), true));
							}
							
							
							default:
							{
								#line 91 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
								return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(-1.6, true), new global::haxe.lang.Null<double>(-3.3, true), new global::haxe.lang.Null<double>(3.2, true), new global::haxe.lang.Null<double>(6.6, true));
							}
							
						}
						
					}
					
					
					default:
					{
						#line 94 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return new global::nx3.geom.Rectangle(new global::haxe.lang.Null<double>(((double) (-2) ), true), new global::haxe.lang.Null<double>(((double) (-2) ), true), new global::haxe.lang.Null<double>(((double) (4) ), true), new global::haxe.lang.Null<double>(((double) (4) ), true));
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_setField(string field, int hash, object @value, bool handleProperties){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				switch (hash){
					case 272654911:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						this.direction = ((global::nx3.EDirectionUD) (@value) );
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return @value;
					}
					
					
					case 1902308268:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						this.notetype = ((global::nx3.ENoteType) (@value) );
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return @value;
					}
					
					
					case 1545099839:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						this.notevalue = ((global::nx3.ENoteVal) (@value) );
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return @value;
					}
					
					
					case 1193672430:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						this.placements = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return @value;
					}
					
					
					case 1389717277:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						this.vheads = ((global::Array<object>) (global::Array<object>.__hx_cast<object>(((global::Array) (@value) ))) );
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return @value;
					}
					
					
					default:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return base.__hx_setField(field, hash, @value, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_getField(string field, int hash, bool throwErrors, bool isCheck, bool handleProperties){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				switch (hash){
					case 933671524:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("headRect") ), ((int) (933671524) ))) );
					}
					
					
					case 325402290:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return ((global::haxe.lang.Function) (new global::haxe.lang.Closure(((object) (this) ), ((string) ("getHeadsRects") ), ((int) (325402290) ))) );
					}
					
					
					case 272654911:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.direction;
					}
					
					
					case 1902308268:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.notetype;
					}
					
					
					case 1545099839:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.notevalue;
					}
					
					
					case 1193672430:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.placements;
					}
					
					
					case 1389717277:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.vheads;
					}
					
					
					default:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return base.__hx_getField(field, hash, throwErrors, isCheck, handleProperties);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   object __hx_invokeField(string field, int hash, global::Array dynargs){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				switch (hash){
					case 933671524:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.headRect(((global::nx3.EHeadType) (dynargs[0]) ), ((global::nx3.ENoteVal) (dynargs[1]) ));
					}
					
					
					case 325402290:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return this.getHeadsRects();
					}
					
					
					default:
					{
						#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
						return base.__hx_invokeField(field, hash, dynargs);
					}
					
				}
				
			}
			#line default
		}
		
		
		public override   void __hx_getFields(global::Array<object> baseArr){
			unchecked {
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				baseArr.push("direction");
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				baseArr.push("notetype");
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				baseArr.push("notevalue");
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				baseArr.push("placements");
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				baseArr.push("vheads");
				#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
				{
					#line 14 "F:\\nx3\\src\\nx3\\PHeadsRectsCalculator.hx"
					base.__hx_getFields(baseArr);
				}
				
			}
			#line default
		}
		
		
	}
}

File: Nx3Java/java/src/nx3/PHeadsRectsCalculator.java
package nx3;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class PHeadsRectsCalculator extends haxe.lang.HxObject
{
	public    PHeadsRectsCalculator(haxe.lang.EmptyObject empty)
	{
		{
		}
		
	}
	
	
	public    PHeadsRectsCalculator(nx3.PNote note, nx3.EDirectionUD direction)
	{
		nx3.PHeadsRectsCalculator.__hx_ctor_nx3_PHeadsRectsCalculator(this, note, direction);
	}
	
	
	public static   void __hx_ctor_nx3_PHeadsRectsCalculator(nx3.PHeadsRectsCalculator __temp_me107, nx3.PNote note, nx3.EDirectionUD direction)
	{
		if (( direction != null )) 
		{
			__temp_me107.direction = direction;
		}
		 else 
		{
			__temp_me107.direction = note.getDirection();
		}
		
		__temp_me107.vheads = note.get_heads();
		__temp_me107.placements = new nx3.PHeadPlacementsCalculator(((haxe.root.Array<nx3.PHead>) (__temp_me107.vheads) ), ((nx3.EDirectionUD) (__temp_me107.direction) )).getHeadsPlacements();
		__temp_me107.notevalue = note.nnote.value;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new nx3.PHeadsRectsCalculator(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new nx3.PHeadsRectsCalculator(((nx3.PNote) (arr.__get(0)) ), ((nx3.EDirectionUD) (arr.__get(1)) ));
	}
	
	
	public  haxe.root.Array<nx3.PHead> vheads;
	
	public  haxe.root.Array<java.lang.Object> placements;
	
	public  nx3.ENoteVal notevalue;
	
	public  nx3.ENoteType notetype;
	
	public  nx3.EDirectionUD direction;
	
	public   haxe.root.Array<nx3.geom.Rectangle> getHeadsRects()
	{
		haxe.root.Array<nx3.geom.Rectangle> rects = new haxe.root.Array<nx3.geom.Rectangle>();
		int i = 0;
		{
			int _g = 0;
			haxe.root.Array<java.lang.Object> _g1 = this.placements;
			while (( _g < _g1.length ))
			{
				java.lang.Object placement = _g1.__get(_g);
				 ++ _g;
				nx3.EHeadType headtype = this.vheads.__get(i).nhead.type;
				nx3.geom.Rectangle rect = null;
				double headw = ((double) (0) );
				this.headRect(headtype, this.notevalue);
				{
					nx3.EHeadValueType _g2 = nx3.ENoteValTools.head(this.notevalue);
					switch (haxe.root.Type.enumIndex(_g2))
					{
						case 2:
						{
							headw = 2.2;
							break;
						}
						
						
						default:
						{
							headw = 1.6;
							break;
						}
						
					}
					
				}
				
				nx3.geom.Rectangle rect1 = new nx3.geom.Rectangle(((java.lang.Object) ( - (headw) ) ), ((java.lang.Object) (-1) ), ((java.lang.Object) (( 2 * headw )) ), ((java.lang.Object) (2) ));
				double pos = 0.0;
				{
					nx3.EHeadPosition _g21 = ((nx3.EHeadPosition) (haxe.lang.Runtime.getField(placement, "pos", true)) );
					switch (haxe.root.Type.enumIndex(_g21))
					{
						case 0:
						{
							pos = ( -2 * headw );
							break;
						}
						
						
						case 2:
						{
							pos = ( 2 * headw );
							break;
						}
						
						
						default:
						{
							pos = ((double) (0) );
							break;
						}
						
					}
					
				}
				
				rect1.offset(pos, ((double) (((int) (haxe.lang.Runtime.getField_f(placement, "level", true)) )) ));
				rects.push(rect1);
				i++;
			}
			
		}
		
		return rects;
	}
	
	
	public   nx3.geom.Rectangle headRect(nx3.EHeadType type, nx3.ENoteVal notevalue)
	{
		int headw = 2;
		int headh = 2;
		switch (haxe.root.Type.enumIndex(type))
		{
			case 0:
			{
				nx3.EHeadValueType _g = nx3.ENoteValTools.head(this.notevalue);
				switch (haxe.root.Type.enumIndex(_g))
				{
					case 2:
					{
						return new nx3.geom.Rectangle(((java.lang.Object) (-2.2) ), ((java.lang.Object) (-1) ), ((java.lang.Object) (4.4) ), ((java.lang.Object) (2) ));
					}
					
					
					default:
					{
						return new nx3.geom.Rectangle(((java.lang.Object) (-1.6) ), ((java.lang.Object) (-1) ), ((java.lang.Object) (3.2) ), ((java.lang.Object) (2) ));
					}
					
				}
				
			}
			
			
			case 3:
			{
				int _g1 = nx3.ENoteValTools.beaminglevel(this.notevalue);
				switch (_g1)
				{
					case 1:
					{
						return new nx3.geom.Rectangle(((java.lang.Object) (-1.8) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (3.6) ), ((java.lang.Object) (6) ));
					}
					
					
					case 2:
					{
						return new nx3.geom.Rectangle(((java.lang.Object) (-2) ), ((java.lang.Object) (-3) ), ((java.lang.Object) (4) ), ((java.lang.Object) (6) ));
					}
					
					
					default:
					{
						return new nx3.geom.Rectangle(((java.lang.Object) (-1.6) ), ((java.lang.Object) (-3.3) ), ((java.lang.Object) (3.2) ), ((java.lang.Object) (6.6) ));
					}
					
				}
				
			}
			
			
			default:
			{
				return new nx3.geom.Rectangle(((java.lang.Object) (-2) ), ((java.lang.Object) (-2) ), ((java.lang.Object) (4) ), ((java.lang.Object) (4) ));
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef696 = true;
			switch (field.hashCode())
			{
				case -962590849:
				{
					if (field.equals("direction")) 
					{
						__temp_executeDef696 = false;
						this.direction = ((nx3.EDirectionUD) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -817575971:
				{
					if (field.equals("vheads")) 
					{
						__temp_executeDef696 = false;
						this.vheads = ((haxe.root.Array<nx3.PHead>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1582102060:
				{
					if (field.equals("notetype")) 
					{
						__temp_executeDef696 = false;
						this.notetype = ((nx3.ENoteType) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -253474258:
				{
					if (field.equals("placements")) 
					{
						__temp_executeDef696 = false;
						this.placements = ((haxe.root.Array<java.lang.Object>) (value) );
						return value;
					}
					
					break;
				}
				
				
				case 1801652415:
				{
					if (field.equals("notevalue")) 
					{
						__temp_executeDef696 = false;
						this.notevalue = ((nx3.ENoteVal) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef696) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef697 = true;
			switch (field.hashCode())
			{
				case -1115837468:
				{
					if (field.equals("headRect")) 
					{
						__temp_executeDef697 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("headRect"))) );
					}
					
					break;
				}
				
				
				case -817575971:
				{
					if (field.equals("vheads")) 
					{
						__temp_executeDef697 = false;
						return this.vheads;
					}
					
					break;
				}
				
				
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef697 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("getHeadsRects"))) );
					}
					
					break;
				}
				
				
				case -253474258:
				{
					if (field.equals("placements")) 
					{
						__temp_executeDef697 = false;
						return this.placements;
					}
					
					break;
				}
				
				
				case -962590849:
				{
					if (field.equals("direction")) 
					{
						__temp_executeDef697 = false;
						return this.direction;
					}
					
					break;
				}
				
				
				case 1801652415:
				{
					if (field.equals("notevalue")) 
					{
						__temp_executeDef697 = false;
						return this.notevalue;
					}
					
					break;
				}
				
				
				case 1582102060:
				{
					if (field.equals("notetype")) 
					{
						__temp_executeDef697 = false;
						return this.notetype;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef697) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			boolean __temp_executeDef698 = true;
			switch (field.hashCode())
			{
				case -1115837468:
				{
					if (field.equals("headRect")) 
					{
						__temp_executeDef698 = false;
						return this.headRect(((nx3.EHeadType) (dynargs.__get(0)) ), ((nx3.ENoteVal) (dynargs.__get(1)) ));
					}
					
					break;
				}
				
				
				case 2051530802:
				{
					if (field.equals("getHeadsRects")) 
					{
						__temp_executeDef698 = false;
						return this.getHeadsRects();
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef698) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("direction");
		baseArr.push("notetype");
		baseArr.push("notevalue");
		baseArr.push("placements");
		baseArr.push("vheads");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


